pkg/registry/generic/etcd/etcd.go

```diff
@@ -127,6 +131,28 @@ func NamespaceKeyFunc(ctx api.Context, prefix string, name string) (string, erro
 	return key, nil
 }
 
+// NoNamespaceKeyRootFunc is the default function for constructing etcd paths to root-scoped resource directories.
+func NoNamespaceKeyRootFunc(ctx api.Context, prefix string) (string, error) {
```

Comment:
So this is the use case that obviously did not exist when we only had namespaced resources in the pre-node all things are namespaced era of the project.  I guess what I don't know is if there is a benefit to pro-actively check for a Namespace on a context if the function completely ignores it.  Was there code that was going direct to the registry layer and not through a client where this caused an impact that warrants changing the signature here versus higher up the stack like you did in rest layer?